agent-inject/agent/config.go

```diff
@@ -81,9 +82,16 @@ func (a *Agent) newTemplateConfigs() []*Template {
 			template = fmt.Sprintf(DefaultTemplate, secret.Path)
 		}
 
+		destination := fmt.Sprintf("/vault/secrets/%s", secret.Name)
+
+		if strings.HasPrefix(secret.Name, "/") {
+			// it is only possible to use an absolute directory location with AnnotationAgentInjectLocation
+			destination = secret.Name
+		}
+
 		tmpl := &Template{
 			Contents:    template,
-			Destination: fmt.Sprintf("/vault/secrets/%s", secret.Name),
+			Destination: destination,
```

Comment:
based on further testing, it seems that there is a bug when the destination name contains a "/". Can someone help me trace this issue? Would it be in the vault agent code or the consul template code?

The issue is that the template does not seem to be applied if I chose a destination with that forward slash, we get the default 
```
{{ with secret \"%s\" }}{{ range $k, $v := .Data }}{{ $k }}: {{ $v }}\n{{ end }}{{ end }}
```
ie the file is created with the right filename containing the "/", but the template is not applied to it.

When I select a destination without "/" the template is applied properly.

Reply:
Also any filename longer than about 20 char gets created, but the same issue where the template does not get applied happens.

Reply:
@tvoran or @jasonodonnell if you could help me trace that issue I could fix it too! I took a look at the vault agent code and cannot find where that length limitation could be. The unit tests include filename lengths that are quite long too so I'm wondering if the issue is somewhere in the integration of vault-k8s to vault-agent?

Reply:
``` json
    {
      "destination": "/Users/smit.thakkar/projects/container_environment/CHAT_DJANGO_SECRET_KEY",
      "contents": "{{- with secret \"secret/beta/chat-service/chat-service-beta-tech/CHAT_DJANGO_SECRET_KEY\" -}} {{ .Data.value }} TEST_SMIT {{- end }}",
      "left_delimiter": "{{",
      "right_delimiter": "}}"
    },
```

I tried above and it worked for me.

Reply:
I don't think filename longer than 20 char has any issue, what version are you using

Reply:
when the destination name contains a "/" I get below error 
2020-03-06T16:49:22.480+0400 [ERROR] template.server: template server error: error="error rendering "(dynamic)" => "/Users/smit.thakkar/projects/container_environment/CHAT_DJANGO_SECRET_KEY/": failed reading file: open /Users/smit.thakkar/projects/container_environment/CHAT_DJANGO_SECRET_KEY/: not a directory"

OR

2020-03-06T16:49:09.889+0400 [ERROR] template.server: template server error: error="error rendering "(dynamic)" => "/Users/smit.thakkar/projects/container_environment/": failed reading file: read /Users/smit.thakkar/projects/container_environment/: is a directory"